### DISCUSSION TITLE
Limited Roborock usability

### DISCUSSION THREAD

**Discussion by JDrillz** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#518] [Q&A] [answered]
Hello, I'm a newbie of HA world, so this will probably be a naive question, btw...
I have a Roborock S8, and I replaced the Roborock app and the Roborock official integration with (respectively) Xiaomi Home and Xiaomi Miio in order to use Xiaomi Cloud Map Extractor; everything works fine, but with the previous Roborock integration I was able to see many more sensors (over 30 vs 17). In particular I've lost the mop sensors and I can't change the amount of water to use. Moreover, I can't manually select rooms to clean.
I'm also using Lovelace Vacuum Map card, and in the "installation manual" I can clearly see a screenshot with the "room mode" available so I cannot understand why I don't have it.
Is the problem this "lack of sensors" of Xiaomi Miio?

Thanks!

Thanks!

**Comment by PiotrMachowski** [comment, marked as answer]
I think it would be better for you to use [this Roborock integration](https://github.com/humbertogontijo/homeassistant-roborock). It works perfectly fine with my card and should provide more features

**Reply by JDrillz** [reply]
Thanks Piotr, that's definitely better, I can hadle the mop and the water now, and I just discovered that I had to click on "generate rooms configuration" in order to enable the rooms cleaning mode. Awesome.

**Reply by PiotrMachowski** [reply]
Great 👍
